src/validation/joi/joi.shared.schemas.ts

```diff
@@ -174,3 +174,8 @@ export const baseDBEntitySchema: ObjectSchema<BaseDBEntity> = objectSchema<BaseD
 export const macAddressSchema = stringSchema.regex(/^([0-9A-Fa-f]{2}[:-]){5}([0-9A-Fa-f]{2})$/)
 
 export const uuidSchema = stringSchema.uuid()
+
+export const DATE_TIME_STRING_REGEX =
+  /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}(?::\d{2})?(?:Z|[+-]\d{2}:\d{2})?$/
+
+export const dateTimeStringSchema = stringSchema.regex(DATE_TIME_STRING_REGEX)
```

Comment:
Not sure if it's possible to use `localDate.isValidString` here instead. But the current solution looks fine by me

Reply:
Another nitpick: I'd move it up to `dateIntervalStringSchema` as they seems to be "relative" schemas

Reply:
> Not sure if it's possible to use localDate.isValidString here instead. But the current solution looks fine by me

I had the exact same thoughts first as we have already date part defined in [js-lib](https://github.com/NaturalCycles/js-lib/blob/ec31b569ecc875d9690c57f00476b11f75700bf9/src/datetime/localDate.ts#L21). But I also thought this is fine, I couldn't come up with nicer solution than this 😄

Reply:
> Not sure if it's possible to use localDate.isValidString here instead

No, validation schemas should be very low-level and raw, so, depending on heavier things is not recommended.

But I left a comment that it should instead **be compatible** with it - give same valid/invalid results 👍